tools/diagnostics/nsdiff

```diff
@@ -0,0 +1,875 @@
+        abs_file = abspath(file)
+        baseName = basename(abs_file)
+        baseNameElements = baseName.split('.')
+        fileName = None
+        fileExtension = None
+        if len(baseNameElements) > 0:
+            fileName = baseNameElements[0]
+        if len(baseNameElements) > 1:
+            fileExtension = ".".join(baseNameElements[1:])
+        return ( dirname(abs_file), baseName, fileName, fileExtension ) 
+
+    def _mkdir_p(self, path):
+        try:
+            os.makedirs(path)
+        except OSError as exc:  # Python >2.5
+            if exc.errno == errno.EEXIST and os.path.isdir(path):
+                pass
+            else:
+                raise
+    
+    def transform_and_scrub(self, sname, sin, sout):
+        """Convert all state data to a *nix diff-able form
+    
+        Keyword arguments:
+        sin -- A directory containing indy-node/plenum state data
+        sout -- A directory in which to write *nix-diff-able/friendly output
+    
+        Detect all LevelDB/RocksDB directories. The LevelDB directory structure
+        is described here:
+            https://github.com/google/leveldb/blob/master/doc/impl.md 
+
+            Each database is represented by a set of files stored in a
+            directory.
+       
+            By default, LevelDB/RocksDB uses a BytewiseComparator, which is the
+            expected outcome when calling initKeyValueStorage to create a
+            LevelDB/RocksDB instance/directory. initKeyValueStorageIntKeys
+            changes the comparator to an IntegerComparator. Therefore, the
+            LevelDB/RocksDB instances/directories may require one or the other.
+            First Attempt to open with initKeyValueStorage. If that throws an
+            error,try again with initKeyValueStorageIntKeys.
+       
+        Transform and scrub data in preparation for *nix diff
+           1. Marshall/Transform data from LevelDB/RocksDB directories to flat
+              files.  Use domain objects where possible to decode key/value
+              pairs.
+           2. Any subset of 'particpants' (nodes) can be used to create
+              state 'signature'(s). Therefore, scrub state_signature LevelDB
+              entries and exclude 'participants' and 'signature' fields;
+              preserving only the 'value' JSON element.
+           3. Scrub the node_info; preserving only the 'view' field.
+           4. Extract the last/latest (view no, ppSeqNo) tuple from the logs
+              emitted by replica.py and node.py. TODO: update this description
+              once we have concrete examples.
+           5. All other files and directories are unaltered.
+        """
+        logger.debug("Transforming and scrubbing {} into {}".format(sin, sout))
+    
+        # The set of filenames that must exist in a directory to be assumed a
+        # LevelDB/RocksDB instance. Will be used in the following for-loop.
+        leveldb_rocksdb_filenames = [ 'CURRENT', 'LOG' ]
```

Comment:
We migrated to Rocks DB now, so Leveldb is not used anymore.

Reply:
The script was written during the transition. Do you think it worth ripping Leveldb? Have we ripped out LevelDB in the node code? Is it no longer possible to run the node with LevelDB? If so, we should rip it out. @ckochenower